hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
+    (setv
+     step (or step n)
+     coll-clones (tee coll n)
+     slices (genexpr (itertools.islice (get coll-clones start) start None step)
```

Comment:
I think this can just be `islice` instead of `itertools.islice` now that we're auto-importing everything from `itertools`. I wrote this before we did that.